go/controller/api/v1alpha1/autogenagent_types.go

```diff
@@ -32,52 +32,59 @@ type AgentSpec struct {
 	// +kubebuilder:validation:MinLength=1
 	SystemMessage string `json:"systemMessage,omitempty"`
 	// +optional
-	ModelConfigRef string `json:"modelConfigRef"`
+	ModelConfig string `json:"modelConfig,omitempty"`
```

Comment:
Why change this? You changed `apiKeySecretName -> apiKeySecretRef`, for consistency shouldn't this also be a `ref`?

Reply:
i kept ApiKeySecretRef beacuse i don't want to confuse people that it's actually their api key (this was also Yuval's feedback originally on that PR)
everywhere else though we don't have `ref` in the name of a reference string `toolServer`, `participant`, etc)